### PR TITLE
Fix NixOS support (again)

### PR DESCRIPTION
**Pull request by ckiee**
The commits have a combination of messages and comments. Should be enough.

- First commit in the series should be documented in the wiki
- Not tested without #1264

### CODE REVIEW

File: alvr/server/build.rs [hunk 1]
```diff
@@ -12,12 +12,34 @@ fn get_ffmpeg_path() -> PathBuf {
...
+        }
     } else {
-        ffmpeg_path
+        base_ffmpeg_path
     }
 }
 
+#[cfg(target_os = "linux")]
+fn cache_ffmpeg_path() {
+    File::create("../../target/alvr_ffmpeg_path")
```
Comment (zarik5) [inline alvr/server/build.rs:37]: Don't assume the position of the target folder. Instead use `afs::target_dir()`. I'm not on Linux, but on my mac/windows machine I configure cargo to generate the target folder in `/tmp`/`%TEMP`.

**Review by zarik5 — CHANGES_REQUESTED** [review]

**Comment by zarik5** [comment]
Will you deal with the wiki edits yourself?

**Comment by ckiee** [comment]
@zarik5: sure (but ADHD brain, so if I remember)

File: alvr/server/build.rs (hunk 1 above)
Comment (ckiee) [reply alvr/server/build.rs:37]: This is `include_str!`'d on ]the other end](https://github.com/ckiee/ALVR/blob/4ca2f2778f46c5583330449ff745dbcae032cdd4/alvr/launcher/src/commands.rs#L218), so I can't call that runtime function in there without a proc macro. Could move it to `build` but I think that'd be just cluttering that instead (and possibly the distributed zips)

Comment (zarik5) [reply alvr/server/build.rs:37]: if the file is temporary use a temp directory (i think xshell or even std has some helper to resolve the system temp folder).

Comment (ckiee) [reply alvr/server/build.rs:37]: Getting the temp directory on the reading side is also not compile-time. It has to be on a constant path.

**Comment by ckiee** [comment]
Made it a draft to ensure it doesn't get merged before CI pass

**Comment by cab404** [comment]
After some tinkering it indeed works on NixOS~

**Comment by cab404** [comment]
Is there a way to get an artifact for android here?

**Comment by ckiee** [comment]
)::

**Pull request closed without merging**